core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -902,6 +902,11 @@ public void run() {
         }
         cancelIdleTimer(/* permanent= */ false);
         enterIdleMode();
+        // New streams may have come in while shutting down resolver and LB policy. If so, restart
+        // processing by exiting idle mode.
+        if (delayedTransport.hasPendingStreams()) {
```

Comment:
Use the inUseAggregator instead, which will fix all the races we were talking about. I'll have a follow-up to change our idleness semantics to ignore in-progress RPCs (ones on real transports). I'll probably discuss it offline.

Reply:
Done